src/anim/controller/anim-blend-tree-2d-cartesian.js

```diff
@@ -39,11 +41,18 @@ class AnimBlendTreeCartesian2D extends AnimBlendTree {
                 result = math.clamp(1.0 - (AnimBlendTreeCartesian2D._pip.dot(pipj) / pipj.lengthSq()), 0.0, 1.0);
                 if (result < minj) minj = result;
             }
-            this._children[i].weight = minj;
+            child.weight = minj;
```

Comment:
comment on 2 lines above this one, not part of this PR directly but just noticed it.

```
AnimBlendTreeCartesian2D._pip = AnimBlendTreeCartesian2D._p.clone().sub(pi);
result = math.clamp(1.0 - (AnimBlendTreeCartesian2D._pip.dot(pipj) / pipj.lengthSq()), 0.0, 1.0);
```
you have those two static variables to avoid allocations, _pip and _p, and here you clone _p and assign it to _pip. It seems this should be written without cloning and simply by using those static variables.

Reply:
I've changed this to use set rather than clone. Also in the process I noticed that I was mistakenly calculating _pip inside the second nested for loop. I've moved it out now :).